Nautilus/Patchers/ProtobufSerializerPrecompiledPatcher.cs

```diff
@@ -0,0 +1,173 @@
+
+        InitializeSerializerEntries(serializeMethodInfo);
+    }
+
+    /// <summary>
+    /// Initialize <see cref="SerializerEntries"/> with Subnautica's known types
+    /// </summary>
+    private static void InitializeSerializerEntries(MethodInfo serializeMethodInfo)
+    {
+        SerializerEntries = new();
+        
+        // We loop through the IL code instructions to find the exact match between a typeId and the method it calls
+        List<KeyValuePair<OpCode, object>> serializeMethodInstructions = PatchProcessor.ReadMethodBody(serializeMethodInfo).ToList();
+        for (int i = 0; i < serializeMethodInstructions.Count; i++)
+        {
+            KeyValuePair<OpCode, object> instruction = serializeMethodInstructions[i];
+            if (instruction.Key == OpCodes.Call)
+            {
+                // The castclass instruction is always 3 steps before the call instruction
+                Type objectType = (Type)serializeMethodInstructions[i - 3].Value;
+                // The ldc.i4 instruction is always 2 steps before the call instruction
+                int associatedTypeId = (int)serializeMethodInstructions[i - 2].Value;
+                // To get the methodName we remove the 5 first letters and we cut before the opening parenthese
+                string serializeMethodName = instruction.Value.ToString()[5..].Split('(')[0];
+                // The deserialize method name will be the same but with Deserialize[...] instead of Serialize[...]
+                string deserializeMethodName = serializeMethodName.Replace("S", "Des");
+
+                SerializerEntries.Add(associatedTypeId, new(objectType, serializeMethodName, deserializeMethodName));
+            }
+        }
+    }
+
+    /// <summary>
+    /// Registers a type as serializable for Protobuf.
+    /// </summary>
+    /// <remarks>
+    /// This is the only method from <see cref="ProtobufSerializerPrecompiledPatcher"/> which should be used by mods.
+    /// </remarks>
+    /// <typeparam name="T">Type of the serializable type</typeparam>
+    /// <param name="typeId">
+    /// An arbitrary unique type id which shouldn't be already used by Subnautica nor by any other mod.
+    /// Look through <see cref="ProtobufSerializerPrecompiled"/> to ensure the id is not already taken.
+    /// </param>
+    /// <param name="serializeMethod">
+    /// A reference to the serialize method.
+    /// NB: Take inspiration from <see cref="ProtobufSerializerPrecompiled.Serialize11492366"/> to understand how make one.
+    /// </param>
+    /// <param name="deserializeMethod">
+    /// A reference to the serialize method.
+    /// NB: Take inspiration from <see cref="ProtobufSerializerPrecompiled.Deserialize11492366"/> to understand how make one.
+    /// </param>
+    public static void RegisterSerializableType<T>(int typeId, Action<T, int, ProtoWriter> serializeMethod, Func<T, ProtoReader, T> deserializeMethod)
```

Comment:
I believe you're missing a handler class. This method is made public, but the patcher class is internal, which makes the method unusable by mods.

The way we have such things setup in Nautilus is by having a public handler under the `Handlers` namespace that subscribes to the patch(es).

Example: [LanguagePatcher](https://github.com/SubnauticaModding/Nautilus/blob/master/Nautilus/Patchers/LanguagePatcher.cs#L97) & [LanguageHander](https://github.com/SubnauticaModding/Nautilus/blob/master/Nautilus/Handlers/LanguageHandler.cs#L22).